d3a_interface/device_validator.py

```diff
@@ -24,10 +25,153 @@
 GeneralSettings = ConstSettings.GeneralSettings
 LoadSettings = ConstSettings.LoadSettings
 PvSettings = ConstSettings.PVSettings
+HomeMeterSettings = ConstSettings.HomeMeterSettings
 StorageSettings = ConstSettings.StorageSettings
 CepSettings = ConstSettings.CommercialProducerSettings
 
 
+class DeviceValidator(metaclass=ABCMeta):
+    """Interface for devices' validator classes."""
+
+    @classmethod
+    def validate(cls, **kwargs):
+        """Validate both rate and energy values of the device."""
+        cls.validate_price(**kwargs)
+        cls.validate_energy(**kwargs)
+
+    @classmethod
+    @abstractmethod
+    def validate_price(cls, **kwargs):
```

Comment:
```suggestion
    def validate_rate(cls, **kwargs):
```

Reply:
I've used the same naming that was being used in the other validators (for load, pv, and storage). I think you're right though, since we're validating rates and not prices. I think the naming should be changed for the other validators too, but I would like to do it in a separate PR (to use validator classes for all of the devices).